src/test/java/org/sagebionetworks/bridge/hibernate/HibernateSurveyDaoTest.java

```diff
@@ -0,0 +1,706 @@
+    }
+    
+    @Test
+    public void getSurveyIncludeElements() throws Exception {
+        HibernateSurvey existing = getHibernateSurvey();
+        
+        when(mockHelper.getById(eq(HibernateSurvey.class), any())).thenReturn(existing);
+        when(mockHelper.queryGet(any(), any(), any(), any(), eq(HibernateSurveyElement.class)))
+            .thenReturn(HIBERNATE_ELEMENTS);
+        
+        GuidCreatedOnVersionHolder keys = new GuidCreatedOnVersionHolderImpl(GUID, CREATED_ON);
+        Survey survey = dao.getSurvey(TEST_STUDY, keys, true);
+        assertEquals(survey.getIdentifier(), IDENTIFIER);
+        assertEquals(survey.getElements().size(), 2);
+        assertSame(existing, survey);
+        
+        verify(mockHelper).getById(eq(HibernateSurvey.class), surveyIdCaptor.capture());
+        assertEquals(surveyIdCaptor.getValue().getSurveyGuid(), GUID);
+        assertEquals(surveyIdCaptor.getValue().getCreatedOn(), CREATED_ON);
+    }
+    
+    @Test
+    public void getSurveyNotFound() {
+        GuidCreatedOnVersionHolder keys = new GuidCreatedOnVersionHolderImpl(GUID, CREATED_ON);
+        
+        assertNull(dao.getSurvey(TEST_STUDY, keys, true));
+    }
+    
+    @Test
+    public void getSurveyWithIdentifier() throws Exception {
+        HibernateSurvey existing = getHibernateSurvey();
+        
+        when(mockHelper.queryGet(any(), any(), any(), any(), eq(HibernateSurvey.class))).thenReturn(ImmutableList.of(existing));
+        when(mockHelper.queryGet(any(), any(), any(), any(), eq(HibernateSurveyElement.class))).thenReturn(HIBERNATE_ELEMENTS);
+        
+        GuidCreatedOnVersionHolder keys = new GuidCreatedOnVersionHolderImpl(IDENTIFIER_PREFIX+IDENTIFIER, CREATED_ON);
+        Survey survey = dao.getSurvey(TEST_STUDY, keys, true);
+        assertEquals(survey.getIdentifier(), IDENTIFIER);
+        assertEquals(survey.getElements().size(), 2);
+        assertSame(existing, survey);
+        
+        verify(mockHelper).queryGet(queryCaptor.capture(),paramsCaptor.capture(), any(), any(), eq(HibernateSurvey.class));
+        
+        assertEquals(queryCaptor.getValue(), "SELECT survey FROM HibernateSurvey as survey WHERE " +
+                "studyKey = :studyKey AND identifier = :identifier AND deleted = 0 AND createdOn = :createdOn " + 
+                "ORDER BY createdOn DESC");
+        assertEquals(paramsCaptor.getValue().get("studyKey"), TEST_STUDY_IDENTIFIER);
+        assertEquals(paramsCaptor.getValue().get("identifier"), IDENTIFIER);
+        assertEquals(paramsCaptor.getValue().get("createdOn"), CREATED_ON);
+    }
+    
+    @Test
+    public void getSurveyNotFoundWithIdentifier() {
+        GuidCreatedOnVersionHolder keys = new GuidCreatedOnVersionHolderImpl(IDENTIFIER_PREFIX+IDENTIFIER, CREATED_ON);
+        
+        assertNull(dao.getSurvey(TEST_STUDY, keys, true));
+    }
+    
+    @Test
+    public void getSurveyAllVersionsIncludeDeleted() {
```

Comment:
What about the exclude deleted version of this test? And the exclude deleted version of the test with identifier?